openmdao/core/_checks.py

```diff
@@ -82,9 +82,9 @@ def _check_types_match(src, tgt, to_prom_name):
     if stype == ttype:
         return
 
-    src_indices = tgt.get('src_indices')
-    if src_indices and len(src_indices) == 1 and ttype == float:
-        return
+    if 'src_indices' in tgt:
+        if len(tgt['src_indices']) == 1 and ttype == float:
+            return
 
```

Comment:
Yeah, I used to do it all the time, but It turns out that using .get() is actually slower, at least in the tests that I've done, than testing for existence and then doing the lookup, especially in cases where most of the time the key doesn't exist.